### PR TITLE
implementacja funkcji usuwania karty

### PR DESCRIPTION
**Pull request by paweldowgiallo84**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The task goal was to delete an existing flashcard from the database when deleting one.

### Requirements
1. Fetch API must be used
2. The following endpoint must be used: `https://training.nerdbord.io/api/v1/fischkapp/flashcards/:id` with a DELETE request and 'Authorization: secret_token' header.

### Code Changes Review

The code changes seem to meet the task requirements. Here's a breakdown:

1. The `fetch` API is used to send a DELETE request to the server, which aligns with the first requirement.

```javascript
fetch(`${FISCHKAPP_URL}/${_id}`, {
    method: 'DELETE',
    headers: {
        'Authorization': URL_AUTH_TOKEN,
    }
})
```

2. The endpoint used in the fetch request is dynamic and depends on the `_id` of the card to be deleted. This aligns with the second requirement.

```javascript
`${FISCHKAPP_URL}/${_id}`
```

3. The 'Authorization' header is included in the fetch request, which is also part of the second requirement.

```javascript
headers: {
    'Authorization': URL_AUTH_TOKEN,
}
```

4. The code handles server response and errors correctly. If the server response is not ok, an error is thrown. If the card is successfully deleted from the server, `setIsDeleting(false)` and `stopEditMode()` are called. If there's an error, it's logged to the console and an error message is set.

```javascript
.then(response => {
    if (!response.ok) {
        throw new Error('Server not responding')
    }
    return response.json()
})
.then(data => {
    console.log('Card deletet from server: ', data)
    setIsDeleting(false)
    stopEditMode()
})
.catch(Error => {
    console.error("Error: ", Error)
    setErrorMsg('Failed to delete card. Please try again.')
})
```

5. The local state of the cards is updated before the fetch request is made. This ensures that the UI is updated immediately, providing a better user experience.

```javascript
const newCards = cards.filter(card => card._id !== _id)
setCards(newCards)
```

In conclusion, the code changes cover the task requirements. However, it's important to note that the actual deletion of the card from the database is not verified in this code review, as it depends on the server's implementation.
